pie/trainer.py

```diff
@@ -284,7 +361,7 @@ def run_check(self, devset):
             print(self.lr_scheduler)
             print()
 
-        return dev_scores
+        return stored_scores
```

Comment:
This bit is getting confusing. We need better naming for dev_scores and stored_scores and also a small docstring explaining what's going on here.